gh-91246: Implement PEP 775: Make zlib required on platforms other than wasi (#130297)
python/cpython: Doc/whatsnew/3.14.rst

* GNU Autoconf 2.72 is now required to generate :file:`configure`.
  (Contributed by Erlend Aasland in :gh:`115765`.)

> * zlib is now required to build CPytho on all platforms except WASI.

Reviewer: ```suggestion
* zlib is now required to build CPython on all platforms except WASI.
```

* GNU Autoconf 2.72 is now required to generate :file:`configure`.
  (Contributed by Erlend Aasland in :gh:`115765`.)

* zlib is now officially required to build CPython.
  It is still technically possible to build CPython without it for special needs, like bootstrapping.
  Such builds are not supported, but we can accept pull requests to keep them working.
  As an exception, zlib is not required on WASI.

>   (Contributed by Stan Ulbrych in :gh:`130297`.)

Reviewer 2: I think you used a lot from Greg's PR.
```suggestion
  (Contributed by Stan Ulbrych and Gregory P. Smith in :gh:`91246`.)
```

> * zlib is now officially required to build CPython.
>   It is still technically possible to build CPython without it for special needs, like bootstrapping.
>   Such builds are not supported, but we can accept pull requests to keep them working.
>   As an exception, zlib is not required on WASI.
>   (Contributed by Stan Ulbrych and Gregory P. Smith in :gh:`91246`.)

Reviewer 3: ```suggestion
* zlib is now required to build CPython on all platforms, except WASI.
  It remains possible to build CPython without zlib,
  which may be useful for scenarios such as bootstrapping.
  Such builds are not supported, but pull requests to keep them working will be accepted.
  (Contributed by Stan Ulbrych and Gregory P. Smith in :gh:`91246`.)
```

Author: Will update these soon
